tests/integration_tests.rs

```diff
@@ -0,0 +1,253 @@
+        assert!(temp.path().join(file).exists(), "{} doesn't exist", file);
+    }
+}
+
+#[test]
+fn rendered_code_has_playpen_stuff() {
+    let temp = create_book(true);
+    let mut md = MDBook::new(temp.path());
+    md.build().unwrap();
+
+    let nested = temp.path().join("book/first/nested.html");
+    let playpen_class = vec![r#"class="playpen""#];
+
+    assert_contains_strings(nested, &playpen_class);
+
+    let book_js = temp.path().join("book/book.js");
+    assert_contains_strings(book_js, &[".playpen"]);
+}
+
+#[test]
+fn chapter_content_appears_in_rendered_document() {
+    let content = vec![
+        ("index.html", "Here's some interesting text"),
+        ("second.html", "Second Chapter"),
+        ("first/nested.html", "testable code"),
+        ("first/index.html", "more text"),
+        ("conclusion.html", "Conclusion"),
+    ];
+
+    let temp = create_book(true);
+    let mut md = MDBook::new(temp.path());
+    md.build().unwrap();
+
+    let destination = temp.path().join("book");
+
+    for (filename, text) in content {
+        let path = destination.join(filename);
+        assert_contains_strings(path, &[text]);
+    }
+}
+
+/// Create a dummy book in a temporary directory, using the contents of
+/// `SUMMARY_MD` as a guide.
+///
+/// The "Nested Chapter" file contains a code block with a single
+/// `assert!($TEST_STATUS)`. If you want to check MDBook's testing
+/// functionality, `$TEST_STATUS` can be substitute for either `true` or
+/// `false`. This is done using the `passing_test` parameter.
+fn create_book(passing_test: bool) -> TempDir {
+    let temp = TempDir::new("dummy_book").unwrap();
+
+    let src = temp.path().join("src");
+    fs::create_dir_all(&src).unwrap();
+
+    File::create(src.join("SUMMARY.md"))
+        .unwrap()
+        .write_all(SUMMARY_MD.as_bytes())
+        .unwrap();
+    File::create(src.join("intro.md"))
+        .unwrap()
+        .write_all(INTRO.as_bytes())
+        .unwrap();
+
+    let first = src.join("first");
+    fs::create_dir_all(&first).unwrap();
+    File::create(first.join("index.md"))
+        .unwrap()
+        .write_all(FIRST.as_bytes())
+        .unwrap();
+
+    let to_substitute = if passing_test { "true" } else { "false" };
+    let nested_text = NESTED.replace("$TEST_STATUS", to_substitute);
+    File::create(first.join("nested.md"))
+        .unwrap()
+        .write_all(nested_text.as_bytes())
+        .unwrap();
+
+    File::create(src.join("second.md"))
+        .unwrap()
+        .write_all(SECOND.as_bytes())
+        .unwrap();
+    File::create(src.join("conclusion.md"))
+        .unwrap()
+        .write_all(CONCLUSION.as_bytes())
+        .unwrap();
+
+    temp
+}
+
+/// Read the contents of the provided file into memory and then iterate through
+/// the list of strings asserting that the file contains all of them.
+fn assert_contains_strings<P: AsRef<Path>>(filename: P, strings: &[&str]) {
+    println!("Checking {}", filename.as_ref().display());
```

Comment:
This is for debugging. Basically when the test fails you won't see the full context of the failure, which in this case would include the file name it failed for, the contents of the file, and the text it's looking for. Normally stdout will get thrown away on a passing test, but when the test fails it'll show you all the information printed out and make identifying why the test failed a lot easier.

I guess you could include those in `assert!()`'s panic message, I've just done it this way out of habit. I can easily swap to printing everything in assert message if you want, should I do that?